### PR TITLE
Blog app: Intergration specs

### PR DESCRIPTION
**Pull request by manzitresor**
### N+1 problem is solved when fetching all posts and their comments for a users screenshoot
users-after
users-before

postBefore
postAfter

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @users = [User.find(params[:user_id])]
+    @users = [User.includes(:posts).find(params[:user_id])]
     @posts = @users[0].posts
```
Comment (afanimartin) [inline app/controllers/posts_controller.rb]: - You are only fetching the User's posts on this line but not post comments as well. Please update this to:
`@posts = @users[0].posts.includes(:comments)`

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @manzitresor, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Repo uses a professional readme template
- I see contributors from both team members.
- `capybara` used for testing
- Tests for User index and show pages added

## Required Changes ♻️
_Check the comments under the review._
- I could not run the tests locally. Please take a look at the screenshot below and fix the errors.
Screenshot from 2023-10-03 12-04-50


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by manzitresor** [comment]
Hi @afanimartin 
I just wanted to ask you the test which are not running is it the because I this project was focusing more on integration tests,
I would appreciate clarifications on the issue.
Thanks

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @manzitresor amd @cosywasswa , 

To highlight 👍🏽 
✔️ No linter errors
✔️ N+1 for posts is fixed


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**